lib/bolt/inventory.rb

```diff
@@ -434,5 +434,17 @@ def invalidate_group_cache!(target)
       @target_plugin_hooks.delete(target.name)
     end
     private :invalidate_group_cache!
+
+    # Logs deprecation warning for Target.new(<uri>, <options>)
+    def target_deprecation
+      return if @deprecation_issued
+
+      @deprecation_issued = true
+      msg = <<~MSG
+        Deprecation Warning: Starting with Bolt 2.0, 'Target.new(<uri>, <options>)' will no
+        longer be supported. Use 'Target.new(<options>)' instead.
```

Comment:
```suggestion
        longer be supported. Use `Target.new(<options>)`, where `options` is a hash with the same structure used to define targets in the inventory V2 file. See https://puppet.com/docs/bolt/latest/writing_plans.html#creating-target-objects for more information.
```

Reply:
We should check `Puppet::Pops::PuppetStack.top_of_stack` to give the file and line number where the warning happened. If we do that, it might be reasonable to issue the warning each time it happens.